docs/tooltip/tooltip.md

```diff
@@ -0,0 +1,212 @@
+# Tooltip Component Specification
+
+A typed React Tooltip component extending the [Popup](../popup/popup.md) component. The tooltip is anchored to another DOM element.
+
+**TBD:**
+* Animation - awaiting API finalization (showAfter - Delay in milliseconds before tooltip is shown, hideAfter - Delay in milliseconds before tooltip is hidden)
+
+## Contents
+
+* [Elements](#elements)
+* [Internal](#internal)
+* [API](#spi)
+* [Style API](#style-api)
+* [Accessibility](#accessibility)
+  * [Roles](#roles)
+  * [Aria Attributes](*aria-attibutes)
+  * [Focus](#focus)
+* [Input Handling](#input-handling)
+  * [Keyboard](#keyboard)
+  * [Mouse](#mouse)
+  * [Touch](#touch)
+* [Examples](#examples)
+
+## Elements
+
+![Image of elements](./assets/tooltipontherange.png)
+
+## Internal
+
+* The tooltip itself is composed of a box and an callout (i.e. arrow). The callout and box are CSS constructs which can be replaced by the user of the component through styling.
+* The arrow always point to the middle of the anchor (in the relevant position), with the box being layed out according to *flipfit* collision rule.
+
+    ![Image of elements](./assets/tooltipfit.png)
+
+    ![Image of elements](./assets/tooltipfit2.png)
+
+## API
+
+| Name | Type | Default | Required | Description |
+| -- | -- | -- | -- | -- |
+| id | string | null | no | Unique identifier |
+| className | string | null | no | Space separated list of the classes of the element |
+| style | Object | null | no | CSS styling declarations to be applied to the element |
+| children | React.ReactNode | null | no | The elements to render in the tooltip |
+| open | boolean | false | no | Whether the tooltip is shown or hidden |
+| onOpenStateChange | function | NOOP | no | Triggered when the popup is opened/closed |
+| anchor | Element | none | Yes | The element to be used as an anchor for the tooltip (will open next to it). |
+| maxHeight | number | 0 | no | The max height in pixels of the popup. If set to 0, the property will be ignored and receive height from its content |
```

Comment:
Shouldn't this be set in the styling via CSS?

Reply:
You can set it via CSS, however need it for TalmiJS...